src/benchmarking.rs

```diff
@@ -17,12 +17,672 @@
+			);
+	}
+
+	report_stake_deposited {
+		let reporter = account::<AccountIdOf<T>>("account", 1, SEED);
+		let address = Address::zero();
+		let amount = trb(100);
+		let caller = deposit_stake::<T>(reporter.clone(), amount, address)?;
+	}: _<RuntimeOrigin<T>>(caller, reporter.clone(), amount, address)
+	verify {
+		assert_last_event::<T>(
+				Event::NewStakerReported { staker: reporter, amount, address }.into(),
+			);
+	}
+
+	report_staking_withdraw_request {
+		let reporter = account::<AccountIdOf<T>>("account", 1, SEED);
+		let address = Address::zero();
+		let amount = trb(100);
+		let caller = deposit_stake::<T>(reporter.clone(), amount, address)?;
+	}: _<RuntimeOrigin<T>>(caller, reporter.clone(), amount, address)
+	verify {
+		let staking_contract = T::Staking::get();
+		assert_last_event::<T>(
+				Event::StakeWithdrawRequestConfirmationSent { para_id: staking_contract.para_id,
+				contract_address: staking_contract.address.into() }.into(),
+			);
+	}
+
+	report_stake_withdrawn {
+		let reporter = account::<AccountIdOf<T>>("account", 1, SEED);
+		let address = Address::zero();
+		let amount = trb(100);
+		let caller = deposit_stake::<T>(reporter.clone(), amount, address)?;
+		// request stake withdraw
+		Tellor::<T>::report_staking_withdraw_request(caller.clone(), reporter.clone(), amount, address)?;
+		T::BenchmarkHelper::set_time(WEEKS);
+	}: _<RuntimeOrigin<T>>(caller, reporter.clone(), amount)
+	verify {
+		assert_last_event::<T>(
+				Event::StakeWithdrawnReported { staker: reporter }.into(),
+			);
+	}
+
+	setup_data_feed {
+		// Maximum value for query data in order to measure the maximum weight
+		let q in 1..T::MaxQueryDataLength::get();
+		let query_data: QueryDataOf<T> = BoundedVec::try_from(vec![1u8; q as usize]).unwrap();
+		let query_id = Keccak256::hash(query_data.as_ref()).into();
+		let feed_creator = account::<AccountIdOf<T>>("account", 1, SEED);
+
+		T::BenchmarkHelper::set_balance(feed_creator.clone(), 1000);
+		// create feed
+		let _ = create_feed::<T>(
+				feed_creator.clone(),
+				query_id,
+				token::<T>(10u64),
+				T::Time::now().as_secs(),
+				700,
+				60,
+				0,
+				token::<T>(0u64),
+				query_data.clone(),
+				token::<T>(1_000u64)
+		);
+
+	}: _(RawOrigin::Signed(feed_creator.clone()), query_id, token::<T>(10u64), T::Time::now().as_secs(), 600, 60, 0, token::<T>(0u64), query_data.clone(), token::<T>(1_000u64))
+
+	fund_feed{
+		let query_data: QueryDataOf<T> = spot_price("dot", "usd").try_into().unwrap();
+		let query_id = Keccak256::hash(query_data.as_ref()).into();
+		let feed_creator = account::<AccountIdOf<T>>("account", 1, SEED);
+
+		T::BenchmarkHelper::set_balance(feed_creator.clone(), 1_000);
+		let feed_id = create_feed::<T>(feed_creator.clone(),
+				query_id,
+				token::<T>(10u64),
+				T::Time::now().as_secs(),
+				700,
+				60,
+				0,
+				token::<T>(0u64),
+				query_data.clone(),
```

Comment:
Please kindly check all usages of .clone() in file, some of them seem unneccessary.